docs/tasks.md

```diff
@@ -371,6 +372,40 @@ For example, use volumes to accomplish one of the following common tasks:
   unsafe_. Use [kaniko](https://github.com/GoogleContainerTools/kaniko) instead.
   This is used only for the purposes of demonstration.
 
+### Workspaces
+
+`workspaces` are a way of declaring volumes you expect to be made available to your
+executing `Task` and the path to make them available at. They are similar to
+[`volumes`](#volumes) but allow you to enforce at runtime that the volumes have
+been attached and [allow you to specify subpaths](taskruns.md#workspaces) in the volumes
+to attach.
+
+The volume will be made available at `/workspace/myworkspace`, or you can or override
+this with `mountPath`. The value at `mountPath` can be anywhere on your pod's filesystem.
+The path will be available via [variable substitution](#variable-substitution) with
+`$(workspaces.myworkspace.path)`.
+
+The actual volumes must be provided at runtime
+[in the `TaskRun`](taskruns.md#workspaces).
+In a future iteration ([#1438](https://github.com/tektoncd/pipeline/issues/1438))
+it [will be possible to specify these in the `PipelineRun`](pipelineruns.md#workspaces)
+as well.
+
+For example:
+
+```yaml
+spec:
+  steps:
+  - name: write-message
+    image: ubuntu
+    command: ['bash']
+    args: ['-c', 'echo hello! > $(workspaces.messages.path)/message']
```

Comment:
kk! the only thing about script mode is i like to have `set -ex` in there when i use it and then it becomes multiple lines :thinking:  but ill do it anyway! sans `set -ex` this time

Reply:
(i guess that's not script mode's fault since either way i dont have `set -ex` - but thanks to script mode i actually thought of it :woman_facepalming: )

Reply:
... i guess i need to do this in workspace.yaml too now

Reply:
... which you literally just said

I HAVE GOT TO STOP SKIMMING

2020 new years resolution

Reply:
+1 to `set -x`.  I'm adding those to all my script mode things.  Makes the log output really super wonderful.  Wonder if it should be the default like how we add the #! right now.

Reply:
https://github.com/tektoncd/pipeline/issues/1735